docs/features/dynamic_macros.md

```diff
@@ -32,12 +32,13 @@ For the details about the internals of the dynamic macros, please read the comme
 
 There are a number of options added that should allow some additional degree of customization
 
-|Define                      |Default         |Description                                                                                                      |
-|----------------------------|----------------|-----------------------------------------------------------------------------------------------------------------|
-|`DYNAMIC_MACRO_SIZE`        |128             |Sets the amount of memory that Dynamic Macros can use. This is a limited resource, dependent on the controller.  |
-|`DYNAMIC_MACRO_USER_CALL`   |*Not defined*   |Defining this falls back to using the user `keymap.c` file to trigger the macro behavior.                        |
-|`DYNAMIC_MACRO_NO_NESTING`  |*Not Defined*   |Defining this disables the ability to call a macro from another macro (nested macros).                           | 
-|`DYNAMIC_MACRO_DELAY`        |*Not Defined*   |Sets the waiting time (ms unit) when sending each key.                                                           |
+|Define                           |Default         |Description                                                                                                      |
+|---------------------------------|----------------|-----------------------------------------------------------------------------------------------------------------|
+|`DYNAMIC_MACRO_SIZE`             |128             |Sets the amount of memory that Dynamic Macros can use. This is a limited resource, dependent on the controller.  |
+|`DYNAMIC_MACRO_USER_CALL`        |*Not defined*   |Defining this falls back to using the user `keymap.c` file to trigger the macro behavior.                        |
+|`DYNAMIC_MACRO_NO_NESTING`       |*Not Defined*   |Defining this disables the ability to call a macro from another macro (nested macros).                           |
+|`DYNAMIC_MACRO_DELAY`            |*Not Defined*   |Sets the waiting time (ms unit) when sending each key.                                                           |
+|`DYNAMIC_MACRO_KEEP_LAYER_STATE` |*Not Defined*   |Defining this keeps the layer state when starting to record a macro                                              |
```

Comment:
`DYNAMIC_MACRO_KEEP_ORIGINAL_LAYER_STATE` might be a bit better of a name as it's clearer about what it is doing.